src/ucp/tag/rndv.c

```diff
@@ -610,6 +617,203 @@ static void ucp_rndv_req_send_rma_get(ucp_request_t *rndv_req, ucp_request_t *rr
     ucp_request_send(rndv_req, 0);
 }
 
+UCS_PROFILE_FUNC_VOID(ucp_rndv_recv_frag_put_completion, (self, status),
+                      uct_completion_t *self, ucs_status_t status)
+{
+    ucp_request_t *freq     = ucs_container_of(self, ucp_request_t,
+                                               send.state.uct_comp);
+    ucp_request_t *req      = freq->send.rndv_put.sreq;
+    ucp_request_t *rndv_req = (ucp_request_t*)freq->send.rndv_put.remote_request;
+
+    ucs_trace_req("freq:%p: recv_frag_put done. rreq:%p ", freq, req);
+
+    /* release memory descriptor */
+    if (freq->send.mdesc != NULL) {
+        ucs_mpool_put_inline((void *)freq->send.mdesc);
+    }
+
+    if (rndv_req != NULL) {
```

Comment:
@bureddy

Reply:
this can be NULL when it is called from ATP handler in put protocol.  there is no rndv_req in that case, wer are just simply PUT the fragment to mem type recv buffer.  whereas in GET pipeline protocol, we are rndv_req and will send ATS after GET all fragments from remote

Reply:
ok, so can you pls comment that in case of put protocol we don't have rndv_req